### PR TITLE
Add using GitHub app authentication in README

### PR DESCRIPTION
**Pull request by Tatsumi0000**
### Before the change?

* There was no explanation in the README for authentication using GitHub App.

### After the change?

* Add using GitHub app authentication for README.

### Other information

Reference information.

* [Authenticating with GitHub Apps](https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps)

----

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

### Pull request type

 
 

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Feature/model/API additions: `Type: Feature`
- Updates to docs or samples: `Type: Documentation`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -298,6 +299,17 @@ user = client.user 'defunkt'
 [access scopes]: http://developer.github.com/v3/oauth/#scopes
 [app-creds]: http://developer.github.com/v3/#increasing-the-unauthenticated-rate-limit-for-oauth-applications
 
+### GitHub App
+Octokit.rb also supports authentication [using GitHub App](https://docs.github.com/en/developers/apps/managing-github-apps/installing-github-apps). 
+
+[Required generated JWT TOKEN](https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-a-github-app).
```
Comment (kfcampbell) [inline README.md:305]:
```suggestion
Octokit.rb also supports authentication [using a GitHub App](https://docs.github.com/en/developers/apps/managing-github-apps/installing-github-apps), which [requires a generated JWT token](https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-a-github-app).
```

Thanks for contributing! I'd like to make a small change to the wording if that's okay.

Comment (Tatsumi0000) [reply README.md:305]: Thanks! 🙆‍♂️

**Review by kfcampbell — APPROVED** [review]
Thank you for contributing! Builds are failing due to an unrelated issue I need to sort out. When I do, this PR will be merged!

**Comment by Tatsumi0000** [comment]
> Thank you for contributing! Builds are failing due to an unrelated issue I need to sort out. When I do, this PR will be merged!

Thanks for review it!
I understand the merge.

**Pull request merged**